ledger/block/src/helpers/target.rs

```diff
@@ -36,12 +36,14 @@ pub fn block_reward<N: Network>(
     time_since_last_block: i64,
     coinbase_reward: u64,
     transaction_fees: u64,
-) -> u64 {
+) -> Result<u64> {
     // Determine which block reward version to use.
-    match block_height < N::CONSENSUS_V2_HEIGHT {
-        true => block_reward_v1(total_supply, block_time, coinbase_reward, transaction_fees),
-        false => block_reward_v2(total_supply, time_since_last_block, coinbase_reward, transaction_fees),
-    }
+    Ok(match block_height {
+        height if height < N::HEIGHT_V(2)? => {
```

Comment:
IMO this code snippet is more confusing than it was before; without context, `height < N::HEIGHT_V(2)` is less clear than `block_height < N::CONSENSUS_V2_HEIGHT`.

It could be the variable naming or the use of a non-constant value of `2`, but it feels a bit unnatural.

Reply:
Maybe a wrapper `enum` would be better than `2`.

For instance: `CONSENSUS::V2`, `CONSENSUS_RULES::V2`, `CONSENSUS_VERSION::V2`.